src/web.go

```diff
@@ -77,72 +77,112 @@ type Web struct {
 	evaluator           *Evaluator
 }
 
+func wildcardRouteHandler(w http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	route, ok := vars["route"]
+
+	if !ok {
+		// TODO Logging -> Local
+		// TODO Logging -> Grafana Loki
+	}
+
+	if mimeType := mime.TypeByExtension(path.Ext(route)); mimeType != "" {
+		w.Header()["Content-Type"] = []string{mimeType}
+	}
+	err := makeViewTemplate(route).Execute(w, vars)
+	if err != nil {
+		makeViewTemplate("index.html").Execute(w, nil)
+		// TODO Logging -> Local
+		// TODO Logging -> Grafana Loki
+		// TODO Update ViewTemplate
+	}
+}
+
+func defaultHandler(w http.ResponseWriter, req *http.Request) {
+	err := makeViewTemplate("index.html").Execute(w, nil)
+	if err != nil {
+		makeViewTemplate("index.html").Execute(w, nil)
+		// TODO Logging -> Local
+		// TODO Logging -> Grafana Loki
+		// TODO Update ViewTemplate
+	}
+}
+
 func (self *Web) start(ctx context.Context) error {
 	self.logger.Println("Starting Web")
 
-	router := bunrouter.New(
-		bunrouter.WithMiddleware(errorHandler),
-	)
-
-	router.GET("/", func(w http.ResponseWriter, req bunrouter.Request) error {
-		return makeViewTemplate("index.html").Execute(w, nil)
-	})
+	router := mux.NewRouter()
+	// TODO Implement Logging to Local
+	// TODO Implement different Log Levels (Info, Warning, Error etc.)
+	// TODO Implement Logging to Loki
+	// TODO Show Errors in viewTemplate
+
+	workflowRouter := router.PathPrefix("/workflow").Subrouter()
+	workflowRouter.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
+		name := req.URL.Query().Get("name")
+
+		var templateName string
+		var instances []*model.WorkflowInstance
+
+		if len(name) == 0 {
+			templateName = "workflow"
+			if insts, err := (*self.workflowService).GetAll(); err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				// TODO Logging -> Local
```

Comment:
Maybe I got confused. But I believe that every local logging is reflected in Loki automatically.

Reply:
There is the LokiQueryRange function in action.go but it only reads Entries from the query_range api endpoint of Loki, so I don't see any writes to Loki happening here.
And there is the NewActionService where a logger is set with some LStdFlags but also no logging to prometheus or Loki directly. @biandratti Can you point me to the code doing the reflected logging to Loki?

Reply:
Is it the right way to use the push endpoints to add logging to Grafana Loki: https://grafana.com/docs/loki/latest/api/ ?

Reply:
We are using logrus as logger dependency. Maybe should be use in every log.
Here you have a way to logger transparently in loki and in your local console.
https://github.com/akkuman/logrus-loki-hook


It should work like a logback appender used in Java to transparently log information to loki.

Reply:
From Working Session: It does log to loki directly via StdOut